### ISSUE TITLE
new hook

### ISSUE THREAD

**Issue by officialmcpro32** [zhuowei/MCPELauncher#470]
@zhuowei 
function blockUpdateHook()
Block.setUpdatable

**Comment by Black-DeathPE** [comment]
Explanation would be nice.

**Comment by ShadowMine64** [comment]
Yrp

**Comment by officialmcpro32** [comment]
function blockUpdateHook(eventType) or add x,y,Z
if(eventType==digBlock && getTile(x,y,Z)==3){ //event types is digBlock && placeBlock

//Block.setUpdatable(id,boolean) //weather the block can be updated

**Comment by Black-DeathPE** [comment]
Doesn't make sense to me and most of that would cause errors anyway

**Comment by Black-DeathPE** [comment]
would need a digBlock constant but the thing is there's destroyBlock(x,y,z, side) which is a better option

**Comment by Black-DeathPE** [comment]
would need a digBlock constant but the thing is there's destroyBlock(x,y,z, side) which is a better option

**Comment by zhuowei** [comment]
When would block update hook get called? Is it only called for player modification of blocks or is it called on any block change (e.g. snow melting, fire spreading)?

**Comment by zhuowei** [comment]
When would block update hook get called? Is it only called for player modification of blocks or is it called on any block change (e.g. snow melting, fire spreading)?

**Comment by officialmcpro32** [comment]
Yes if there a block update nearby it types would be like fireSpread place block and you would specify what should it do if updated @zhuowei
